### PR TITLE
Proposed EPF discontinuity fix

### PR DESCRIPTION
**Pull request by veluca93**
This change to EPF changes a step function into a very high slope linear
segment. After fixing the wrong handling of sigma on ARM, this reduces
the discrepancy between the platforms to roughly the same as what you'd
get without EPF at all.

The difference in decoded pixels is minimal, with changes in BPP*pnorm
of an OOM of 10^-4 even with epf3.

```
Compr              Input    Compr            Compr       Compr  Decomp  Butteraugli
Method            Pixels     Size              BPP   #    MP/s    MP/s     Distance    Error p norm           BPP*pnorm   Errors
--------------------------------------------------------------------------------------------------------------------------------
jxl:d0.7        13270449  3376020    2.03521071518   1   3.330  26.712   1.48137546   0.48043308217  0.9777825567553876        0
jxl:d1          13270449  2668442    1.60865212624   1   3.519  28.528   1.77268112   0.61341852300  0.9867770112915698        0
jxl:d2          13270449  1693685    1.02102649277   1   3.733  29.602   2.93441439   0.96843094388  0.9887936501245655        0
jxl:d4          13270449  1022946    0.61667604465   1   3.652  20.601   4.81405544   1.56954856488  0.9679030008724722        0
jxl:d8          13270449   583016    0.35146723370   1   3.535  19.850   9.03337669   2.58973218211  0.9102060060710633        0
Aggregate:      13270449  1555270    0.93758378970 ---   3.551  24.717   3.19915795   1.03014322850  0.9658455921082402        0

Compr              Input    Compr            Compr       Compr  Decomp  Butteraugli
Method            Pixels     Size              BPP   #    MP/s    MP/s     Distance    Error p norm           BPP*pnorm   Errors
--------------------------------------------------------------------------------------------------------------------------------
jxl:d0.7        13270449  3376020    2.03521071518   1   3.318  26.453   1.48137546   0.48042693781  0.9777700516912733        0
jxl:d1          13270449  2668442    1.60865212624   1   3.500  28.507   1.77268100   0.61340812266  0.9867602807648902        0
jxl:d2          13270449  1693685    1.02102649277   1   3.711  29.250   2.93475580   0.96840029958  0.9887623614804250        0
jxl:d4          13270449  1022946    0.61667604465   1   3.638  20.212   4.81493282   1.56972404547  0.9680112155454044        0
jxl:d8          13270449   583016    0.35146723370   1   3.524  19.400   9.04940987   2.59019622279  0.9103691011673687        0
Aggregate:      13270449  1555270    0.93758378970 ---   3.536  24.400   3.20048384   1.03019052919  0.9658899404742294        0
```

### CODE REVIEW

File: lib/jxl/epf.cc [hunk 1]
```diff
@@ -55,19 +55,26 @@ using DU = HWY_CAPPED(uint32_t, GroupBorderAssigner::kPaddingXRound);
 // kInvSigmaNum / 0.3
 constexpr float kMinSigma = -3.90524291751269967465540850526868f;
 
+namespace {
+constexpr float kSlope = 64.0f;
```
Comment (jonsneyers) [inline lib/jxl/epf.cc:59]: Why 64?
I suppose a power of two has advantages, but other than that, why 2^6 and not 2^5 or 2^7?

Comment (veluca93) [reply lib/jxl/epf.cc:59]: 128 is too much (it ~doubles the error), 64 is the highest power of two that gets the same error as no EPF (and thus the one that changes the results the least)

**Review by jonsneyers — APPROVED** [review]

**Comment by jonsneyers** [comment]
Any reason not to merge this one? (will have to rebase to get the new CI to run)

**Comment by jonsneyers** [comment]
Closing as it will be superseded by https://github.com/libjxl/libjxl/pull/1204

**Pull request closed without merging**